frontend/src/metabase/admin/performance/components/StrategyForm.tsx

```diff
@@ -164,6 +186,29 @@ const StrategyFormBody = ({
     }
   }, [selectedStrategyType, values, setFieldValue]);
 
+  useEffect(
```

Comment:
I really love to avoid syncing state / using `useEffect` as much as possible, I find less bugs result from deriving values / using `useMemo` when possible. My question is, instead of syncing the state to formik so that it computes dirtiness, would it be possible for us to compute dirtiness on our own by taking the formik idea of dirtiness + detecting if values are set to their defaults.